lib/configuration.js

```diff
@@ -15,150 +15,430 @@
+  /**
+   * The _projectId property is meant to contain the string project id that the
+   * hosting application is running under. The project id is a unique string
+   * identifier for the project. If the Configuration instance is not able to
+   * retrieve a project id from the metadata service or the runtime-given
+   * configuration then the property will remain null. If given both a project
+   * id through the metadata service and the runtime configuration then the
+   * instance will assign the value given by the metadata service over the
+   * runtime configuration. If the instance is unable to retrieve a valid
+   * project id or number from runtime configuration and the metadata service
+   * then this will trigger the `error` event in which listening components must
+   * operate in 'offline' mode.
+   * {@link https://cloud.google.com/compute/docs/storing-retrieving-metadata}
+   * @memberof Configuration
+   * @private
+   * @type {String|Null}
+   * @defaultvalue null
+   */
+  this._projectId = null;
+  /**
+   * The _projectNumber property is meant to contain the number project id that
+   * the hosting application is running under. The project number is a unique
+   * number identifier for the project. If the Configuration instance is unable
+   * to retrieve a project number from the metadata service or the runtime
+   * configuration then the property will remain null. If given both a project
+   * number through the metadata service and the runtime configuration then the
+   * instance will assign the value given by the metadata service over the
+   * runtime configuration. If the Configuration instance is not able to 
+   * retrieve a project number or a project id from runtime configuration or
+   * from the metadata service then this will trigger the `error` event in which
+   * listening components must operate in 'offline' mode.
+   * {@link https://cloud.google.com/compute/docs/storing-retrieving-metadata}
+   * @memberof Configuration
+   * @private
+   * @type {String|Null}
+   * @defaultvalue null
+   */
+  this._projectNumber = null;
+  /**
+   * The _key property is meant to contain the optional Stackdriver API key that
+   * may be used in place of default application credentials to authenticate
+   * with the Stackdriver Error API. This property will remain null if a key
+   * is not given in the runtime configuration or an invalid type is given as
+   * the runtime configuration.
+   * {@link https://support.google.com/cloud/answer/6158862?hl=en}
+   * @memberof Configuration
+   * @private
+   * @type {String|Null}
+   * @defaultvalue null
+   */
+  this._key = null;
+  /**
+   * The _serviceContext property is meant to contain the optional service
+   * context information which may be given in the runtime configuration. If
+   * not given in the runtime configuration then the property value will remain
+   * null.
+   * @memberof Configuration
+   * @private
+   * @type {Object}
+   * @default
```

Comment:
@matthewloring supposed to be `@defaultvalue {Object}` (its null right now but that will change with the update to this pr today). Thanks!